impl/src/main/java/eu/darkbot/impl/galaxy/SpinResultImpl.java

```diff
@@ -0,0 +1,183 @@
+package eu.darkbot.impl.galaxy;
+
+import eu.darkbot.api.game.galaxy.GalaxyGate;
+import eu.darkbot.api.game.galaxy.SpinResult;
+import eu.darkbot.api.game.items.SelectableItem;
+import eu.darkbot.util.XmlUtils;
+import org.w3c.dom.Element;
+
+import java.time.Instant;
+import java.util.*;
+import java.util.stream.Stream;
+
+public class SpinResultImpl implements SpinResult {
+    private final GalaxyInfoImpl galaxyInfo;
+
+    private final Map<ItemType, SpinInfoImpl> itemsResult = new EnumMap<>(ItemType.class);
+    private final Map<SelectableItem.Laser, SpinInfo> ammoResult;
+
+    private int parts, multipliers;
+
+    private Instant date;
+    private GalaxyGate gate;
+
+    SpinResultImpl(GalaxyInfoImpl galaxyInfo) {
+        this.galaxyInfo = galaxyInfo;
+
+        Map<SelectableItem.Laser, SpinInfo> ammoResult = new HashMap<>();
+
+        for (ItemType item : ItemType.values()) {
+            SelectableItem.Laser laser = item.laser;
+            if (laser != null) ammoResult.put(laser, getSpinInfo(item));
+        }
+
+        this.ammoResult = Collections.unmodifiableMap(ammoResult);
```

Comment:
```suggestion
        this.ammoResult = Collections.unmodifiableMap(
                ItemType.values().stream()
                    .filter(it -> it.laser != null)
                    .collect(Collectors.toMap(it -> it.laser, this::getSpinInfo)));
```
This can be replaced by a simple stream